### PR TITLE
Camera intrinsics - First phase of #148

### PR DESCRIPTION
**Pull request by 104H**

### CODE REVIEW

**Comment by 104H** [comment]
First phase of #148

File: seerep-srv/seerep-server/src/fb-cameraintrinsics-service.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+grpc::Status FbCameraIntrinsicsService::GetCameraIntrinsics(
+    grpc::ServerContext* context, const flatbuffers::grpc::Message<seerep::fb::cameraIntrinsicsQuery>* request,
+    grpc::ServerWriter<flatbuffers::grpc::Message<seerep::fb::CameraIntrinsics>>* writer)
+{
+  (void)context;  // ignore this variable without causing warnings
+  auto requestRoot = request->GetRoot();
+
+  // TODO check both project and camera intrinsics uuid are set
+
+  seerep_core_msgs::camera_intrinsics_query ciq_coremsg;
```
Comment (Mark-Niemeyer) [inline seerep-srv/seerep-server/src/fb-cameraintrinsics-service.cpp:18]: please use try-catch in both services to catch potential errors. Check e.g. fb-image-service.cpp

File: seerep-hdf5/seerep-hdf5-core/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,7 @@ include(FindHighFive)
 
 find_package(SeerepMsgs REQUIRED)
 find_package(HighFive REQUIRED)
+find_package(gRPC CONFIG REQUIRED)
```
Comment (Mark-Niemeyer) [inline seerep-hdf5/seerep-hdf5-core/CMakeLists.txt:14]: Do we really need grpc here? The core should be agnostic...

**Review by Mark-Niemeyer — CHANGES_REQUESTED** [review]
please also provide a small python script to test the set and get grpc services

File: seerep-srv/seerep-server/src/fb-cameraintrinsics-service.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  ciq_coremsg = seerep_core_fb::CoreFbConversion::fromFb(*requestRoot);
+
+  seerep_core_msgs::camera_intrinsics ci;
+  ciCore->getData(ciq_coremsg, ci);
```
[the remark this answers is not in the document]
Comment (104H) [reply seerep-srv/seerep-server/src/fb-cameraintrinsics-service.cpp:22]: Is the right approach to convert the core msg `ci` to a FB CI and then write it to the writer?

File: seerep-srv/seerep-core-fb/src/core-fb-camera-intrinsics.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#include "seerep-core-fb/core-fb-camera-intrinsics.h"
+
+namespace seerep_core_fb
+{
+CoreFbCameraIntrinsics::CoreFbCameraIntrinsics(std::shared_ptr<seerep_core::Core> seerepCore) : m_seerepCore(seerepCore)
+{
+  // CoreFbGeneral::getAllFileAccessorFromCore(m_seerepCore, m_hdf5IoMap);
```
Comment (Mark-Niemeyer) [inline seerep-srv/seerep-core-fb/src/core-fb-camera-intrinsics.cpp:7]: remove?

File: seerep-msgs/fbs/region_of_interest.fbs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+
+namespace seerep.fb;
+
+table regionOfInterest
```
Comment (Mark-Niemeyer) [inline seerep-msgs/fbs/region_of_interest.fbs:4]: Start table name with upper case for consistency

File: examples/python/gRPC/util_fb.py [hunk 6]
```diff
@@ -23,8 +24,10 @@
     TimeInterval,
     Timestamp,
     TransformStampedQuery,
+    cameraIntrinsicsQuery,
 )
 from fb import meta_operations_grpc_fb as metaOperations
+from fb import regionOfInterest
```
Comment (Mark-Niemeyer) [inline examples/python/gRPC/util_fb.py:30]: why is this separately and not in line 3-27?

**Review by Mark-Niemeyer — CHANGES_REQUESTED** [review]

**Comment by 104H** [comment]
This pull request was carried forward after a rebase in #273 .

**Pull request closed without merging**